### PR TITLE
Add tests for notices view

### LINKED ISSUE #44: Add basic tests for HTML interface for editing notices
We should nail down the endpoints and query parameters of them.

The admin test suite for allow-list should give a good idea how this can be done. https://github.com/ssb-ngi-pointer/go-ssb-room/blob/f774817536669c4f2ce2fe356a172d047ef5e780/web/handlers/admin/allow_list_test.go#L109

Check that

* [x] admidb/sqlite: the default notices are present (`/notice?id=x` for different x's) 
* [x] web/handlers: the _edit_ buttons show for logged in admins
* [x] web/handlers/admin: the _edit_ form includes all the needed fields, for input=hidden and input=text
  * [x] for add translation (new)
  * [x] for edit (existing)
* [x] web/handlers/admin: the _save_ handlers refuse incomplete requests (all the params included)
* [x] web/handlers/admin: make sure the _save_ handler calls the mockdb  (CallCount and CallArgs)

### PR DESCRIPTION
**Pull request by cblgh**
This PR closes #44, adding tests for various parts of the notices functionality & view.

### CODE REVIEW

File: web/utils.go [hunk 1]
```diff
@@ -40,7 +40,7 @@ func NewURLTo(appRouter *mux.Router) func(string, ...interface{}) *url.URL {
 	return func(routeName string, ps ...interface{}) *url.URL {
 		route := appRouter.Get(routeName)
 		if route == nil {
-			level.Warn(l).Log("msg", "no such route", "route", routeName, "params", fmt.Sprintf("%v", ps))
+			// level.Warn(l).Log("msg", "no such route", "route", routeName, "params", fmt.Sprintf("%v", ps))
```
Comment (cblgh) [inline web/utils.go:43]: gotta remember to uncomment this before merging :)

Comment (cryptix) [reply web/utils.go:43]: If you could mark it with a link to #35 _that would be great_.

Comment (cblgh) [reply web/utils.go:43]: mark how, by adding a link to #35 in the src?

File: web/handlers/admin/notices_test.go [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+	id := []string{"1"}
+	title := []string{"SSB Breaking News: This Test Is Great"}
+	content := []string{"Absolutely Thrilling Content"}
+	language := []string{"en-GB"}
+
+	// POST a correct request to the save handler, and verify that the save was handled using the mock database)
+	u := urlTo(router.AdminNoticeSave)
+	formValues := url.Values{"id": id, "title": title, "content": content, "language": language}
+	resp := ts.Client.PostForm(u.String(), formValues)
+	a.NotEqual(http.StatusInternalServerError, resp.Code)
```
Comment (cryptix) [inline web/handlers/admin/notices_test.go:32]: NotEqual is nice but Equal would better. Could be bad request for instance.

**Review by cryptix — CHANGES_REQUESTED** [review]
Really like what i'm seeing here!

One big-ish request though: rebase this on master, please and change the other uses of the renamed `webasserts.ElementsInForm`

**Comment by cblgh** [comment]
i think this is ok to merge @cryptix :)

**Review by cryptix — APPROVED** [review]
Push the button!

**Comment by cblgh** [comment]
ahhhhhhhhhhhhhhhhhhh

image

**Comment by cryptix** [comment]
Horray!

ps: usually we delete the old branches. Not sure when we wouldn't actually.

**Comment by cblgh** [comment]
> Horray!
> 
> ps: usually we delete the old branches. Not sure when we wouldn't actually.

gotcha! thanks for the reminder, i typically forget about that and do spring cleaning on all the old branches when i come across them haha

**Pull request merged**